### PR TITLE
Fix thread deadlock issue

### PR DESCRIPTION
**Pull request by kcudnik**
Add separate lock for notification and reuse
global syncd lock to protect only db access

### CODE REVIEW

File: syncd/syncd.cpp [hunk 1]
```diff
@@ -148,6 +148,8 @@ void remove_rid_and_vid_from_local(
 sai_object_id_t translate_rid_to_vid(
         _In_ sai_object_id_t rid)
 {
+    std::lock_guard<std::mutex> lock(g_mutex);
```
Comment (yxieca) [inline syncd/syncd.cpp]: Does it make sense to rename g_mutex to g_db_mutex?

Comment (kcudnik) [inline syncd/syncd.cpp]: yes

File: syncd/syncd_counters.cpp [hunk 2]
```diff
@@ -8,7 +8,11 @@ void collectCounters(swss::Table &countersTable,
     // collect counters should be under mutex
     // sice configuration can change and we
     // don't want that during counters collection
-    std::lock_guard<std::mutex> lock(g_mutex);
+    //
+    // XXX currently lock for counters is disabled
+    // since on new 0.9.4 can cause deadlock since
+    // interanl sai implementaion also uses locks
+    // std::lock_guard<std::mutex> lock(g_db_mutex);
```
Comment (lguohan) [inline syncd/syncd_counters.cpp]: >g_db_mutex [](start = 40, length = 10)

can we move this lock to line 35 so that we avoid the lock for the sai api?

Comment (kcudnik) [inline syncd/syncd_counters.cpp]: sure

File: syncd/syncd.cpp (hunk 1 above)
Comment (yxieca) [inline syncd/syncd.cpp]: Thanks Buddy!

**Pull request merged**